File: PlacementCellBackend/Services/CRUD/Interfaces/IAlumniService.cs
using PlacementCellBackend.DTOs.AlumniDtos;
using PlacementCellBackend.Models;

namespace PlacementCellBackend.Services.CRUD.Interfaces;

public interface IAlumniService
{
    Task<IEnumerable<AlumniDto>> GetAllAlumniAsync();
    Task<AlumniDto?> GetAlumniByIdWithCompanyAsync(string id);
    Task<Alumni?> GetAlumniByIdAsync(string id);
    Task<Alumni> CreateAlumniAsync(Alumni alumni);
    Task<bool> UpdateAlumniAsync(string id, AlumniDtoUpdate alumni);
    Task<bool> DeleteAlumniAsync(string id);
    bool AlumniExists(string id);
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/CRUD/Interfaces/IAlumniService.java
package com.example.placementicsbackend.services.CRUD.Interfaces;

import com.example.placementicsbackend.dtos.AlumniDtos.AlumniDto;
import com.example.placementicsbackend.dtos.AlumniDtos.AlumniDtoUpdate;
import com.example.placementicsbackend.models.Alumni;
import java.util.List;

public interface IAlumniService {

    List<AlumniDto> getAllAlumni();

    AlumniDto getAlumniByIdWithCompany(String id);

    Alumni getAlumniById(String id);

    Alumni createAlumni(Alumni alumni);

    boolean updateAlumni(String id, AlumniDtoUpdate alumni);

    boolean deleteAlumni(String id);

    boolean alumniExists(String id);
}
